rust/lance-io/src/object_store.rs

```diff
@@ -755,10 +755,14 @@ async fn configure_store(url: &str, options: ObjectStoreParams) -> Result<Object
             base_path: Path::from(url.path()),
             block_size: 64 * 1024,
         }),
-        s => Err(Error::IO {
-            message: format!("Unsupported URI scheme: {}", s),
-            location: location!(),
-        }),
+        unknow_scheme => {
+            // TODO: Check the correct way to construct a object_store::Error::Generic
+            let err = lance_core::Error::from(object_store::Error::Generic {
+                store: "",
+                source: format!("Unsupported URI scheme: {}", unknow_scheme).into(),
+            });
```

Comment:
I'd maybe go with `ObjectStore::NotSupported` here instead:

```suggestion
            let err = lance_core::Error::from(object_store::Error::NotSupported {
                source: format!("Unsupported URI scheme: {}", unknow_scheme).into(),
            });
```

Reply:
sorry, I forget this two TODO comments, it's removed now, thank you!

Reply:
I think you missed my actual change here, which is to use `Error::NotSupported` instead of `Error::Generic` for this spot. Same with the other comment.

Reply:
ha, I totally missed your comment, fixed.